skrifa/src/outline/glyf/mod.rs

```diff
@@ -827,25 +937,267 @@ impl<'a> Scaler<'a> {
+                contours,
+            };
+            let deltas = self
+                .memory
+                .deltas
+                .get_mut(..point_count + PHANTOM_POINT_COUNT)
+                .ok_or(InsufficientMemory)?;
+            let iup_buffer = self
+                .memory
+                .iup_buffer
+                .get_mut(..point_count + PHANTOM_POINT_COUNT)
+                .ok_or(InsufficientMemory)?;
+            if deltas::simple_glyph(
+                gvar,
+                glyph_id,
+                self.coords,
+                self.outlines.has_var_lsb,
+                glyph,
+                iup_buffer,
+                deltas,
+            )
+            .is_ok()
+            {
+                for (point, delta) in points.iter_mut().zip(deltas) {
+                    *point += *delta;
+                }
+            }
+        }
+        // Apply scaling
+        if self.is_scaled {
+            let scale = self.scale.to_f32();
+            for point in points.iter_mut() {
+                *point = point.map(|c| c * scale);
+            }
+        }
+
+        if points_start != 0 {
+            // If we're not the first component, shift our contour end points.
+            for contour_end in contours.iter_mut() {
+                *contour_end += points_start as u16;
+            }
+        }
+        Ok(())
+    }
+
+    fn load_composite(
+        &mut self,
+        glyph: &CompositeGlyph,
+        glyph_id: GlyphId,
+        recurse_depth: usize,
+    ) -> Result<(), DrawError> {
+        use DrawError::InsufficientMemory;
+        let scale = self.scale.to_f32();
+        // The base indices of the points for the current glyph.
+        let point_base = self.point_count;
+        // Compute the per component deltas. Since composites can be nested, we
+        // use a stack and keep track of the base.
+        let mut have_deltas = false;
+        let delta_base = self.component_delta_count;
+        if self.outlines.gvar.is_some() && !self.coords.is_empty() {
+            let gvar = self.outlines.gvar.as_ref().unwrap();
+            let count = glyph.components().count() + PHANTOM_POINT_COUNT;
+            let deltas = self
+                .memory
+                .composite_deltas
+                .get_mut(delta_base..delta_base + count)
+                .ok_or(InsufficientMemory)?;
+            if deltas::composite_glyph(gvar, glyph_id, self.coords, &mut deltas[..]).is_ok() {
+                // If the font is missing variation data for LSBs in HVAR then we
+                // apply the delta to the first phantom point.
+                if !self.outlines.has_var_lsb {
+                    self.phantom[0].x += deltas[count - 4].x;
+                }
+                have_deltas = true;
+            }
+            self.component_delta_count += count;
+        }
+        if self.is_scaled {
+            for point in self.phantom.iter_mut() {
+                *point *= scale;
+            }
+        }
```

Comment:
Same as simple. I'm not sure where this goes

Reply:
Deferred to https://github.com/googlefonts/fontc/issues/784